File: android-stable/app/src/main/java/com/suhas/multyfiautobuy/stable/BootReceiver.java
package com.suhas.multyfiautobuy.stable;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;

public final class BootReceiver extends BroadcastReceiver {
    @Override
    public void onReceive(Context context, Intent intent) {
        String action = intent == null ? "" : intent.getAction();
        if (Intent.ACTION_BOOT_COMPLETED.equals(action)
                || Intent.ACTION_LOCKED_BOOT_COMPLETED.equals(action)
                || Intent.ACTION_USER_UNLOCKED.equals(action)
                || Intent.ACTION_MY_PACKAGE_REPLACED.equals(action)) {
            StrategyMonitorService.ensureRunning(context);
        }
    }
}

File: fresh-global-edge/v122/inspection/java/com/suhas/ucsentinel/worker/BootReceiver.kt
package com.suhas.globaledgeai.worker

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import com.suhas.globaledgeai.diagnostics.DiagnosticLog

class BootReceiver: BroadcastReceiver() {
    override fun onReceive(context: Context, intent: Intent?) {
        DiagnosticLog.log(context,"BOOT","Received ${intent?.action.orEmpty()}; starting market scanner")
        runCatching { MarketScanService.start(context) }
            .onFailure { DiagnosticLog.log(context,"BOOT","Unable to start market scanner",it) }
    }
}
